### ISSUE TITLE
[a11y]: Check boxes state in charts is announced incorrectly

### ISSUE THREAD

**Issue by preranagiri22** [carbon-design-system/carbon#19018]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

"@carbon/charts-react": "^1.23.0"

### Automated testing tool and ruleset

JAWS_Firefox

### Assistive technology

JAWS_Firefox

### Description

Issue description:

- Though the Bars are visually presented - Connected Check Box controls are showing “Un checked state “ (visually Not Ticked )- Here Visual status indication of the Check Box and connected Bar’s visual availability is not matching
- In both Ticked / Not Ticked state (Visually ), screen reader announced these Check Box states as “Checked – Here , the visual status is not presented rightly to screen reader users .

Recommended fix: Ensure that the visual state of the checkbox (checked or unchecked) accurately reflects the state announced by the screen reader users

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example

NA

### Steps to reproduce

NA

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
Hi, this issue has been moved to the Carbon Charts repository for better tracking.
